### PR TITLE
feat: csm metrics

### PR DESCRIPTION
**Pull request by vgorkavenko**

### CODE REVIEW

File: src/modules/csm/csm.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 
 from src.constants import TOTAL_BASIS_POINTS, UINT64_MAX
 from src.metrics.prometheus.business import CONTRACT_ON_PAUSE
+from src.metrics.prometheus.csm import CSM_CURRENT_FRAME_RANGE_L_EPOCH, CSM_CURRENT_FRAME_RANGE_R_EPOCH, \
```
Comment (madlabman) [inline src/modules/csm/csm.py:8]: :!black %

File: src/modules/csm/state.py [hunk 2]
```diff
@@ -6,6 +6,8 @@
 from pathlib import Path
 from typing import Self
 
+from src.metrics.prometheus.csm import CSM_UNPROCESSED_EPOCHS_COUNT, CSM_NETWORK_AVG_PERFORMANCE, \
```
Comment (madlabman) [inline src/modules/csm/state.py:9]: black

File: src/modules/csm/state.py [hunk 3]
```diff
@@ -141,7 +143,10 @@ def is_empty(self) -> bool:
     def unprocessed_epochs(self) -> set[EpochNumber]:
         if not self._epochs_to_process:
             raise ValueError("Epochs to process are not set")
-        return set(self._epochs_to_process) - self._processed_epochs
+        diff = set(self._epochs_to_process) - self._processed_epochs
+        CSM_UNPROCESSED_EPOCHS_COUNT.set(len(diff))
+        CSM_MIN_UNPROCESSED_EPOCH.set(min(diff) if diff else 0)
+        return diff
```
Comment (madlabman) [inline src/modules/csm/state.py:149]: I think we should do that in the outer scope.

File: src/modules/csm/state.py [hunk 4]
```diff
@@ -150,7 +155,9 @@ def is_fulfilled(self) -> bool:
     @property
     def avg_perf(self) -> float:
         """Returns average performance of all validators in the cache"""
-        return self.network_aggr.perf
+        avg = self.network_aggr.perf
+        CSM_NETWORK_AVG_PERFORMANCE.set(avg)
+        return avg
```
Comment (madlabman) [inline src/modules/csm/state.py:160]: This is kinda useless, I think.

File: src/metrics/prometheus/csm.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+)
+
+CSM_CURRENT_FRAME_RANGE_R_EPOCH = Gauge(
+    "csm_current_frame_range_r_epoch",
+    "Right epoch of the current frame range",
+    namespace=PROMETHEUS_PREFIX,
+)
+
+
+CSM_FINALIZED_EPOCH = Gauge(
```
Comment (F4ever) [inline src/metrics/prometheus/csm.py:19]: We have `ORACLE_SLOT_NUMBER`. Do we need to duplicate?

File: src/modules/csm/csm.py [hunk 6]
```diff
@@ -292,6 +296,10 @@ def current_frame_range(self, blockstamp: BlockStamp) -> tuple[EpochNumber, Epoc
         l_epoch = converter.get_epoch_by_slot(SlotNumber(l_ref_slot + 1))
         r_epoch = converter.get_epoch_by_slot(r_ref_slot)
 
+        # Update Prometheus metrics
+        CSM_CURRENT_FRAME_RANGE_L_EPOCH.set(l_epoch)
```
Comment (F4ever) [inline src/modules/csm/csm.py:300]: Maybe make one metric with two labels?

**Review by F4ever — APPROVED** [review]

**Pull request merged**